test/images/agnhost/connect/connect.go

```diff
@@ -114,11 +120,21 @@ func connectSCTP(dest string, timeout time.Duration) {
 	errCh := make(chan error)
 
 	go func() {
-		conn, err := sctp.DialSCTP("sctp", nil, addr)
-		if err == nil {
-			conn.Close()
+		conn, err = sctp.DialSCTP("sctp", nil, addr)
+		defer func() {
+			errCh <- conn.Close()
+		}()
+		if err != nil {
+			errCh <- err
+		}
```

Comment:
you have to defer only if the connection was successful, you have to check the error first